src/main/java/com/example/springboot/controller/HelloController.java

```diff
@@ -1,24 +1,48 @@
 package com.example.springboot.controller;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.example.springboot.ShoppingListService;
+import com.example.springboot.service.Product;
 
 @RestController
+@RequestMapping("/api/v1/item")
 public class HelloController {
+	List<Product> ls = new ArrayList<Product>();
+	
 	@Autowired
 	ShoppingListService sls;
 	
-	@GetMapping("/getShop")
-	public int getShop(int n) {
-		return sls.incrementValue(n);
+	@DeleteMapping
+	public int removeProduct(@RequestParam("name")String name) {
+		for(Product pr:ls) {
+			if(pr.getName()==name) {
+				ls.remove(pr);
+				return 200;
+			}
+		}
+		return 404;
 	}
-	
-	@GetMapping("/")
-	public String index() {
-		return "Greetings from Spring Boot!";
+	@GetMapping
+	public List<Product> showList() {
+		return ls;
+	}
+	@PostMapping
+	public int addProduct(Product p) {
+		if(!ls.contains(p)) {
+			ls.add(p);
+			return 200;
```

Comment:
200 and 400 are HTTP status codes, not "response bodies".
By default, those methods return a 200. To return other code:

- Change the return type of the function to `ResponseEntity<Type>`.
E.g. `ResponseEntity<Integer>` (primitive types must be replaced by their wrapper class), `ResponseEntity<Void>` (No value) or `ResponseEntity<?>` (Any value)
- Instead of the value, return `ResponseEntity.ok().build()` for a 200 empty code (OK = 200), `ResponseEntity.ok(object)` for a 200 with a body, or `ResponseEntity.badRequest().build()` for a empty 400 (Bad request = 400)